packages/babel-plugin-transform-react-constant-elements/src/index.js

```diff
@@ -11,15 +11,39 @@ export default function () {
         return;
       }
 
+      // Elements with refs are not safe to hoist.
       if (path.isJSXIdentifier({ name: "ref" }) && path.parentPath.isJSXAttribute({ name: path.node })) {
         return stop();
       }
 
+      // Ignore identifiers & JSX expressions.
       if (path.isJSXIdentifier() || path.isIdentifier() || path.isJSXMemberExpression()) {
         return;
       }
 
-      if (!path.isImmutable()) stop();
+      if (!path.isImmutable()) {
+        // If it's not immutable, it may still be a pure expression, such as string concatenation.
+        // It is still safe to hoist that, so long as its result is immutable.
+        // If not, it is not safe to replace as mutable values (like objects) could be mutated after render.
+        // https://github.com/facebook/react/issues/3226
+        if (path.isPure()) {
```

Comment:
pure template literals such as  
```
<a className={`ok`}/>
```
is missed

Reply:
@yiminghe isnt it covered already by `isPure` [here](https://github.com/babel/babel/blob/99be60b53d39e26a761b2de3a04b677ff7c5d929/packages/babel-traverse/src/scope/index.js#L632)?

Reply:
no, not covered

Reply:
the lastest version seems to have template literals check, while the latest babel-traverse@6.26.0 does not  @Andarist